LibraBFT/Impl/Consensus/Liveness/ProposerElection.agda

```diff
@@ -22,7 +22,14 @@ isValidProposerM : Author → Round → LBFT Bool
 isValidProposer : ProposerElection → Author → Round → Bool
 
 isValidProposalM : Block → LBFT Bool
-isValidProposalM b = maybeS (b ^∙ bAuthor) (pure false) (λ a → isValidProposerM a (b ^∙ bRound))
+isValidProposalM b = do
```

Comment:
What about using something like `maybeS-RWST`, which has the same type signature as `maybeS` (so it would read like the implementation, with the additional `-RWST` suffix), but which uses the branching operation of the `RWST` datatype directly?